compiler/parser/parser.cpp

```diff
@@ -934,6 +934,30 @@ static ModuleSymbol* dynoParseFile(const char* fileName,
 
   if (dynoRealizeErrors()) USR_STOP();
 
+
+  if (fDynoSerialize) {
+
+    if (strcmp(dynoBinAstDir, "") != 0) {
+      auto sfname = builderResult.serializeToDir(dynoBinAstDir);
+      if (fVerify) {
+        // 'res' = AstList for now - eventually will be a BuilderResult
+        auto result = chpl::uast::BuilderResult::deserializeFromFile(gContext, sfname);
+
+        if (builderResult.compare(result) == false) {
+          USR_FATAL("FAILED TO (DE)SERIALIZE %s\n", builderResult.filePath().c_str());
+        }
+      }
+    } else if (fVerify) {
+      // unspecified output directory, verify only
+      std::stringstream ss;
+      builderResult.serializeToDir(ss);
+      auto res = chpl::uast::BuilderResult::deserializeFromFile(gContext, ss);
+      if (builderResult.compare(res) == false) {
```

Comment:
I anticipate this duplicated code not sticking around very long, so I'd lean towards leaving it as-is and see how the implementation develops in the near future rather than come up with an elegant template function. Does that work for you?

Reply:
Yep, though maybe I'd add a TODO or something to indicate that the duplication should eventually be removed just to "lock down" this discussion in the code :)